website/docs/help-and-support/troubleshooting-guide/action-errors/datasource-errors.md

```diff
@@ -66,14 +66,32 @@ At least one of the mandatory fields in the plugin's datasource creation form is
 
 This message indicates that one of the mandatory fields, for example, `Access Key`, has been left empty in the [datasource creation form](../../../core-concepts/connecting-to-data-sources/connecting-to-databases.md).
 
-This error can be fixed by filling the mentioned mandatory field(s) in the [datasource creation form](../../../core-concepts/connecting-to-data-sources/connecting-to-databases.md).
+This error can be fixed by filling the mentioned mandatory fields in the [datasource creation form](../../../core-concepts/connecting-to-data-sources/connecting-to-databases.md).
 
-### Cannot Delete Datasource Error
+### Can't delete datasource error
 
 ```
 Cannot delete datasource since it has 1 action(s) using it.
 ```
 
-This message indicates that the [datasource](../../../reference/datasources/) attempting to be deleted has some query action configured on it.
+This message indicates that the [datasource](/reference/datasources/) attempting to be deleted has some query action configured on it.
 
-This error can be fixed by deleting any queries dependent on this [datasource](../../../reference/datasources/) before attempting to delete the [datasource](../../../reference/datasources/).
+This error can be fixed by deleting any queries dependent on this [datasource](/reference/datasources/) before attempting to delete the [datasource](../../../reference/datasources/).
+
+### Error connecting to local db or API
+
+If you are trying to connect to a local MySQL database from Appsmith and see an error message like:
+
+ ```
+ io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: /172.17.0.1:3306
```

Comment:
I see that for all the other errors, we show an example of the _user-visible error message_, but this one, I think, is an errors message from the _server logs_, and not visible to the user. Can we use the user-visible error message here as well. We can include this in addition too, of course. What say?

Reply:
"**Error - Connection refused**"- users will receive this error(_visible_) message, right? @sharat87

Reply:
Hey, sorry man, I'm unfortunately not familiar with the error message that shows up. You can try producing this error on Appsmith by deliberately connecting to an invalid address, and use that perhaps. :)

Reply:
Tried this on cloud, this is what we get a
![Screenshot 2022-12-26 at 7 07 40 PM](https://user-images.githubusercontent.com/1757421/209554735-e2314620-a763-4c2c-a26b-9a40a97dcd7e.png)
t the moment: